### PR TITLE
Add Advanced Create to select image 🖼 of VM 🖥

### PR DESCRIPTION
**Pull request by nturinski**
Fixes https://github.com/microsoft/vscode-azurevirtualmachines/issues/98
Fixes https://github.com/microsoft/vscode-azurevirtualmachines/issues/107
Partially fixes https://github.com/microsoft/vscode-azurevirtualmachines/issues/2

Right now we're still limited to Linux VMs.  I think the next step will to include the Windows images listed in the Portal

image

After that, I'll implement the size.

The images are hard-coded for now.  I was investigating for a while, but I wasn't able to find a way to use the API to get the "recommended" picks from the Portal.  Furthermore, the virtualMachineImages API didn't return a human-readable string for the display name so using the API to retrieve the vmImage seemed somewhat pointless.

I ended up just creating a bunch of VMs from the Portal and looked at what their OS configurations were set to to hardcore these values.  Jing is trying to get me in contact with someone in the VS team that dealt with creating VMs through VS so hopefully I can figure out a more dynamic solution.

### CODE REVIEW

File: src/commands/createVirtualMachine/ImageListStep.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    private getAvailableImages(): ImageReferenceWithLabel[] {
+        return [
+            {
+                label: 'Ubuntu Server 18.04 LTS - Gen1',
+                publisher: 'Canonical',
+                offer: 'UbuntuServer',
+                sku: '18.04-LTS',
+                version: 'latest',
+                exactVersion: '18.04.202007290'
```
Comment (EricJizbaMSFT) [inline src/commands/createVirtualMachine/ImageListStep.ts:35]: This "exactVersion" property makes me nervous - could get out of date quickly. Is it necessary?

File: src/tree/SubscriptionTreeItem.ts [hunk 2]
```diff
@@ -86,8 +87,12 @@ export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
         if (promptForPassphrase) {
             promptSteps.push(new PassphrasePromptStep());
         }
-        LocationListStep.addStep(wizardContext, promptSteps);
 
+        if (context.advancedCreation) {
+            promptSteps.push(new ImageListStep());
```
Comment (EricJizbaMSFT) [inline src/tree/SubscriptionTreeItem.ts:92]: I'd prefer if you did:
```
if (!context.advancedCreation) {
    wizardContext.image = // whatever image is default
}
```

And added `ImageListStep` to the list of steps regardless of if it's `advancedCreation` or not. It's just a more flexible/readable way of doing the defaults.

File: src/commands/createVirtualMachine/ImageListStep.ts (hunk 1 above)
Comment (nturinski) [reply src/commands/createVirtualMachine/ImageListStep.ts:35]: I think we can take this out, yeah.  What actually gets used in the linuxConfiguration is just "version" which uses the latest.

File: src/tree/SubscriptionTreeItem.ts [hunk 3]
```diff
@@ -97,6 +99,12 @@ export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
         executeSteps.push(new VirtualMachineCreateStep());
 
         const title: string = 'Create new virtual machine';
+
+        if (!context.advancedCreation) {
+            // for basic create, default to image Ubuntu 18.04 LTS, the first on the list
+            wizardContext.image = new ImageListStep().getAvailableImages()[0];
```
Comment (EricJizbaMSFT) [inline src/tree/SubscriptionTreeItem.ts:105]: Oof do not like this line

Just make a const like this and reference it both here and inside `getAvailableImages`
```
export const ubuntu1804Image: ImageReferenceWithLabel = {
    label: 'Ubuntu Server 18.04 LTS - Gen1',
    publisher: 'Canonical',
    offer: 'UbuntuServer',
    sku: '18.04-LTS',
    version: 'latest'
}
```

Comment (nturinski) [reply src/tree/SubscriptionTreeItem.ts:105]: Coward!

**Pull request merged**